public/components/event_analytics/explorer/explorer.tsx

```diff
@@ -545,19 +545,57 @@ export const Explorer = ({
                         <EuiSpacer size="m" />
                       </>
                     )}
-                    <DataGrid
-                      http={http}
-                      pplService={pplService}
-                      rows={explorerData.jsonData}
-                      rowsAll={explorerData.jsonDataAll}
-                      explorerFields={explorerFields}
-                      timeStampField={queryRef.current![SELECTED_TIMESTAMP]}
-                      rawQuery={appBasedRef.current || queryRef.current![RAW_QUERY]}
-                      totalHits={_.sum(countDistribution.data?.['count()'])}
-                      requestParams={requestParams}
-                      startTime={appLogEvents ? startTime : dateRange[0]}
-                      endTime={appLogEvents ? endTime : dateRange[1]}
-                    />
+                    {isDefaultDataSourceType || appLogEvents ? (
+                      <>
+                        {countDistribution.data && countDistribution.data['count()'] ? (
```

Comment:
question: if countDistribution doesn't contain data or data['count()'], will it be stuck on loading documents?

Reply:
It will be, should I do something else for this then?

Reply:
I don't think we have existing code logics to really wait for countdistribution data fetching, I think either you want to implement that as well to make it waiting on the async fetching and remove the loading component whenever we receive the the result regardless of success or error, or it you don't have enough time to do it i think it's fine to remove it for now

Reply:
removed